polkaj-schnorrkel/src/rust/src/merlin_jni/mod.rs

```diff
@@ -0,0 +1,54 @@
+use robusta_jni::bridge;
+
+pub use self::jni::TranscriptData;
+
+#[bridge]
+pub mod jni {
+  use robusta_jni::convert::{
+    Field, Signature, TryFromJavaValue,
+  };
+  use robusta_jni::jni::objects::AutoLocal;
+
+  #[derive(Signature, TryFromJavaValue)]
+  #[package(io.emeraldpay.polkaj.merlin)]
+  pub struct TranscriptData<'env: 'borrow, 'borrow> {
+      #[allow(dead_code)]
+      #[instance]
+      raw: AutoLocal<'env, 'borrow>,
+
+      //TODO:
+      //  Figure out how to make this simply a `Box<[u8]>` instead of a `Vec<Box<[u8]>>`,
+      //  currently a blocker with byte[] fields (seems to be an issue with robusta itself, some missing impls?)
+      #[field]
+      pub domainSeparationLabel: Field<'env, 'borrow, Vec<Box<[u8]>>>,
+
+      #[field]
+      pub labels: Field<'env, 'borrow, Vec<Box<[u8]>>>,
+
+      #[field]
+      pub messages: Field<'env, 'borrow, Vec<Box<[u8]>>>,
+  }
+}
+
+// TODO:
+//  Try to avoid having to `Box::leak`,
+//  In theory, should be safe enough if used correctly Java-side, since "we know Java owns the object"
```

Comment:
That's also a topic I'm rather unsure about. Leaking the memory (effectively telling rust "leave it up to me to handle") should be okay since the memory is actually owned and managed by the Java runtime. I am, however, lacking the deepest understanding as of what possible issues we could have with that type of memory management :D

Reply:
After digging into this for a little bit, I managed to convince myself that's rather OK. Proper documentation has been added on the intentionality of this piece of code.